src/Integration.Vsix.UnitTests/Suppression/RoslynSuppressionHandlerTests.cs

```diff
@@ -101,134 +99,25 @@ public void ShouldReport_CannotCreateLiveIssue_ReturnsTrue()
             result.Should().BeTrue();
         }
 
-        [TestMethod]
-        public void ShouldReport_NoServerIssues_ReturnsTrue()
+        [DataTestMethod]
+        [DataRow(false, true)]
+        [DataRow(true, false)]
+        public void ShouldReport_MatchingSuppressionExists(bool suppressionExists, bool expectedShouldIssueBeReported)
         {
             // Arrange
             Diagnostic diag = CreateDiagnostic("dummy rule id", CreateSourceLocation());
             SetLiveIssue(diag, startLine: 1, wholeLineText: "text");
-            SetServerIssues(Array.Empty<SonarQubeIssue>());
+            SetSuppressionExistsResponse(suppressionExists);
 
-            RoslynSuppressionHandler handler = new RoslynSuppressionHandler(issueFactoryMock.Object, issueProviderMock.Object);
+            RoslynSuppressionHandler handler = new RoslynSuppressionHandler(issueFactoryMock.Object, issueMatcherMock.Object);
 
             // Act
             bool result = handler.ShouldIssueBeReported(syntaxTreeMock.Object, diag);
 
             // Assert
             VerifyLiveIssueCreated(Times.Once());
             VerifyServerIssuesRequested(Times.Once());
-            result.Should().BeTrue();
-        }
-
-        [TestMethod]
-        public void ShouldReport_NoMatchingServerIssues_ReturnsTrue()
```

Comment:
The tests are much simpler since they are now only checking for a true/false from the IssueMatcher.

These tests were replaced the the IssueMatcherTests in the previous commit.